.github/workflows/ci.yml

```diff
@@ -1,7 +1,104 @@
+########################################################################################################
+# Build configurations
+# ====================
+#
+# full
+# ----
+# Description: Basic build with all components (TLX, C++, Cython). Has to be tested on every fully 
+#              supported platform. Close to user-experience, except that warnings are also treated 
+#              as errors during build and TLX is provided as externally build library (normal 
+#              installation builds both NetworKit and TLX in one step).
+# Script:      scripts/full.sh
+# Build-type:  Release
+#
+# 
+# full (variant)
+# ---------------------
+# Description: These builds are variants of full with different options.
+# Script:      scripts/full.sh
+# Variants:    native = Build for native architecture
+# 
+#         
+# core
+# ----
+# Description: Builds the C++ code into one shared library. Has also sanity checks enabled.
+# Script:      scripts/core.sh
+# Build-type:  Debug
+#
+# 
+# core (variant)
+# ---------------------
+# Description: These builds are variants of core with different options. 
+# Script:      scripts/core.sh
+# Variants:    release = Release-build
+#              non-monolithic = Each C++ module is a library
+#              C++17 = Check for C++17 conformance
+#              C++20 = Check for C++20 conformance
+#              native = Build for native architecture
+# 
+#
+# clang-tidy
+# ----------
+# Description: Uses clang-tidy for several static analysis checks. These involve bug prone programming
+#              errors, possible modernizations and performance optiomizations.
+# Script:      scripts/clang_tidy.sh
+# Config:      .clang-tidy
+# Build-type:  Debug
+#
+# 
+# coverage
+# --------
+# Description: Enables code-coverage, address- & leak-sanitizers and tests in order to create cpp-coveralls data.
+# Script:      scripts/core_sanitizers_coverage.sh
+# Build type:  Debug
+# 
+#
+# documentation:
+# --------------
+# Description: Builds documentation of both C++ and Python-code. The result is automatically deployed
+#              when a pull request is merged.
+# Script:      scripts/documentation.sh 
+# Build-type:  Debug
+#
+#
+# Jobs:
+# =====
+# 0. Rationale:
+# - Main platform is Linux, which is therefore used for most of the tests. Basic user experience (full) 
+#   should be tested on all platforms  with the default compiler.
```

Comment:
What does "on all platforms" mean here if we previously only refer to Linux?

Reply:
Should refer to all supported platforms. Restructured this point to make it clearer.